File: backend/src/main/java/turip/tripcourse/controller/dto/response/TripCourseResponse.java
package turip.tripcourse.controller.dto.response;

import turip.place.controller.dto.response.PlaceResponse;
import turip.place.domain.Place;
import turip.tripcourse.domain.TripCourse;

public record TripCourseResponse(
        Long id,
        int visitDay,
        int visitOrder,
        PlaceResponse place
) {
    public static TripCourseResponse from(TripCourse tripCourse, Place place) {
        return new TripCourseResponse(
                tripCourse.getId(),
                tripCourse.getVisitDay(),
                tripCourse.getVisitOrder(),
                PlaceResponse.from(place)
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/content/place/dto/TripCourseResponse.kt
package com.on.turip.data.content.place.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class TripCourseResponse(
    @SerialName("id")
    val id: Long,
    @SerialName("place")
    val place: PlaceResponse,
    @SerialName("visitDay")
    val visitDay: Int,
    @SerialName("visitOrder")
    val visitOrder: Int,
    @SerialName("timeLine")
    val timeLine: String,
)
